.github/workflows/testing_pr.yml

```
@@ -6,29 +6,102 @@ on:
       - "master"
 
 jobs:
-  build:
+  prepare_matrix: ##############################################################################
+    runs-on: ubuntu-latest
+    outputs:
+       matrix_unittest: ${{ steps.setmatrix.outputs.matrix1 }}
+       matrix_tutorialtest: ${{ steps.setmatrix.outputs.matrix2 }}
+
+    steps:
+        - uses: actions/checkout@v2
+
+        - name: Set up Python
+          uses: actions/setup-python@v4.5.0
+
+        - name: Set Dynamic Matrix
+          id: setmatrix
+          run: |
+           matrix_str=$(python utils/info2json.py testing_matrix)
+           echo "matrix1=$matrix_str" >> $GITHUB_OUTPUT
+           matrix_str=$(python utils/info2json.py tutorial_testing_matrix)
+           echo "matrix2=$matrix_str" >> $GITHUB_OUTPUT
+           
+  check_matrix: ##############################################################################
+    runs-on: ubuntu-latest
+    needs: prepare_matrix
+    steps:
+      - name: Install json2yaml
+        run: |
+          sudo npm install -g json2yaml
+      - name: Check matrix definition
+        run: |
+          matrix='${{ needs.prepare_matrix.outputs.matrix }}'
+          echo $matrix
+          echo $matrix | jq .
+          echo $matrix | json2yaml
+
+  unittest: ##############################################################################
```

I propose to use the underscore like in `check_matrix`:
```suggestion
  unit_test: ##############################################################################
```